lib/spack/spack/relocate.py

```diff
@@ -192,7 +192,7 @@ def get_filetype(path_name):
     """
     file = Executable('file')
     file.add_default_env('LC_ALL', 'C')
-    output = file('-b', '-h', '%s' % path_name,
+    output = file('-bh', '%s' % path_name,
```

Comment:
@scheibelp This is just supposed to output 
file -b -h libmathtext.a 
current ar archive
but lib/libmathtext.a appeared in the text file list which makes me think this returned output like only -h was passed
file -h  lib/libmathtext.a 
lib/libmathtext.a: current ar archive

I know the which('file') allows you to pass multiple arguments. Does Executable('file') have different behavior?

Reply:
You should be able to pass multiple arguments to the exe `__call__` invocation. Both the original and replacement invocation should work. They appear to in my case:

```
bash-4.1$ spack-python
Spack version 0.10.0
Python 2.7.10, Linux x86_64
>>> import spack.util.executable as exe
>>> x = exe.Executable('file')
>>> x('/usr/bin/wget')
/usr/bin/wget: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.18, stripped
>>> x('-b', '/usr/bin/wget')
ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.18, stripped
>>> x('-bh', '/usr/bin/wget')
ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.18, stripped
>>> x('-b', '-h', '/usr/bin/wget')
ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 2.6.18, stripped
```

Does the above return something different for you?

Reply:
I think I figure out what the problem was. The problem was with a symbolic link to libmathtext.a. File return 'symbolic link to /path/to/lib/libmathtext. I need to skip any file type that has link it it.